### PR TITLE
Add support for Oasis Scan API v2 as a backend option for Wallet 

### PR DESCRIPTION
**Pull request by buberdds**
https://api.oasisscan.com/v2/swagger/

Testing requires .env update
```
REACT_APP_BACKEND=oasisscanV2
```

Notes:
- will be used as secondary Wallet API   
- v2 API was down for a few days
- API won't provide full historical data
- runtime tx `from` address is oasis1 not eth address compared to v1 https://github.com/bitcat365/oasisscan-backend/issues/13
- tx amount is not rounded anymore in API

TODOs:
- [x] debug why generated API files had to be updated manually.  https://github.com/oasisprotocol/wallet/pull/2075/commits/291fbd30d89fca63943e6a5580645b196396f2ad is needed due to https://github.com/bitcat365/oasisscan-backend/issues/12
- [x] pending tx notification issue related to missing historical data: 
   - oasis1qz6323u2tgpaz8nhxefh9nfrwtfsq4224u5f5cfl account with 0 txs, but with nonce
   - oasis1qpp70ru9g3tw6zxm9te08dz4yfq9mdgu0cz220ta account with partial history with nonce and !noncesFromTxs 
- [x] `runtime` query param will be added to V2 `transactions` endpoint in the future. At this point we cannot show ParaTime deposits or withdraws https://github.com/bitcat365/oasisscan-backend/issues/8
- [x] issue with spec vs API type mismatch https://github.com/bitcat365/oasisscan-backend/issues/9
- [x] incorrect swagger spec https://github.com/bitcat365/oasisscan-backend/issues/7#event-14823468676
- [x] define Allowance enum in swagger https://github.com/bitcat365/oasisscan-backend/issues/10#event-14853377973

### CODE REVIEW

File: src/vendors/oasisscan-v2.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    if (cacheId in getTransactionCacheMap) {
+      return getTransactionCacheMap[cacheId]
+    }
+
+    const transaction = await chainApi.chainTransactionInfoHandler({
+      hash,
+    })
+
+    if (transaction) {
+      getTransactionCacheMap[cacheId] = transaction
```
Comment (lukaw3d) [inline src/vendors/oasisscan-v2.ts:93]: this is nicer than v1

File: src/vendors/oasisscan-v2.ts [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    })
+    if (!transactionsList) throw new Error('Wrong response code')
+
+    const list = await Promise.all(
+      transactionsList.list.map(async tx => {
+        if (tx.txType === 'runtime') {
+          return await getRuntimeTransactionInfo(tx)
+        } else {
+          const { nonce } = await getTransaction({ hash: tx.txHash! })
+          return { ...tx, nonce }
```
[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/vendors/oasisscan-v2.ts:114]: Something is causing a permanent "Some transactions are currently in a pending state." at http://localhost:3000/account/oasis1qrtyn2q78jv6plrmexrsrv4dh89wv4n49udtg2km

and when switching to testnet; "Couldn't load transactions. Unknown error: Cannot destructure property 'nonce' of '(intermediate value)' as it is null."

[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/vendors/oasisscan-v2.ts:114]: Ah, oasisscan v2 is failing again

File: src/vendors/oasisscan-v2.ts [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+  return {
+    address: account.address,
+    allowances: account.allowances.map(allowance => ({
+      address: allowance.address,
+      amount: parseRoseStringToBaseUnitString(allowance.amount),
+    })),
+    available: parseRoseStringToBaseUnitString(account.available),
+    delegations: parseRoseStringToBaseUnitString(account.escrow),
+    debonding: parseRoseStringToBaseUnitString(account.debonding),
+    total: parseRoseStringToBaseUnitString(account.total),
```
[the remark this answers is not in the document]
Comment (lukaw3d) [reply src/vendors/oasisscan-v2.ts:164]: seems good.

It is correct for a
- normal account http://localhost:3000/account/oasis1qz78ap0456g2rk7j6rmtvasc9v2kjhz2s58qgj90
- validator http://localhost:3000/account/oasis1qq3xrq0urs8qcffhvmhfhz4p0mu7ewc8rscnlwxe

**Review by lukaw3d — APPROVED** [review]

File: src/vendors/oasisscan-v2.ts [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+        level: undefined,
+        status: t.result ? TransactionStatus.Successful : TransactionStatus.Failed,
+        timestamp: t.timestamp * 1000,
+        to: t.ctx.to ?? undefined,
+        type: t.ctx.method as TransactionType,
+        runtimeName: t.runtimeName,
+        runtimeId: t.runtimeId,
+        round: t.round,
+        nonce:
+          t.ctx?.nonce ?? t.etx?.nonce != null ? BigInt(t.ctx?.nonce ?? t.etx?.nonce).toString() : undefined,
```
[the remark this answers is not in the document]
Comment (buberdds) [reply src/vendors/oasisscan-v2.ts:208]: was following v1 https://github.com/oasisprotocol/wallet/commit/e9fdf1f3e, but in v2 ctx nonce is not optional so it looks like we can simplify this. I did not find any tx with etx so far.

**Pull request merged**